pyvista/plotting/lookup_table.py

```diff
@@ -0,0 +1,754 @@
+            # this creates a new shallow copy and is necessary to update the
+            # internal VTK array
+            self.table.Get().values = self
+
+    def __array_wrap__(self, out_arr, context=None):
+        """Return a numpy scalar if array is 0d.
+
+        See https://github.com/numpy/numpy/issues/5819
+
+        """
+        if out_arr.ndim:
+            return np.ndarray.__array_wrap__(self, out_arr, context)
+
+        # Match numpy's behavior and return a numpy dtype scalar
+        return out_arr[()]
+
+    __getattr__ = _vtk.VTKArray.__getattr__
+
+
+@no_new_attr
+class LookupTable(_vtk.vtkLookupTable):
+    """Scalar to RGBA mapping table.
+
+    A lookup table is an array that maps input values to output values. When
+    plotting data over a dataset, it is necessary to map those scalars to
+    colors (in the RGBA format), and this class provides the functionality to
+    do so.
+
+    See `vtkLookupTable
+    <https://vtk.org/doc/nightly/html/classvtkLookupTable.html>`_ for more
+    details regarding the underlying VTK API.
+
+    Parameters
+    ----------
+    cmap : str, colors.Colormap, optional
+        Color map from ``matplotlib``, ``colorcet``, or ``cmocean``.
+
+    n_values : int, default: 256
+        Number of colors in the color map.
+
+    flip : bool, default: False
+        Flip the direction of cmap. Most colormaps allow ``*_r`` suffix to do this
+        as well.
+
+    Examples
+    --------
+    Plot the lookup table with the default VTK color map.
+
+    >>> import pyvista as pv
+    >>> lut = pv.LookupTable()
+    >>> lut  # doctest:+SKIP
+    LookupTable (0x7ff3de60d580)
+      Table Range:                (0.0, 1.0)
+      N Values:                   256
+      Above Range Color:          None
+      Below Range Color:          None
+      NAN Color:                  Color(name='maroon', hex='#800000ff')
+      Log Scale:                  False
+      Color Map:                  "VTK lookup table"
+        Alpha Range:              (1.0, 1.0)
+        Hue Range:                (0.0, 0.66667)
+        Saturation Range          (1.0, 1.0)
+        Value Range               (1.0, 1.0)
+        Ramp                      s-curve
+        Is Opaque                 True
+    >>> lut.plot()
+
+    Plot the lookup table with the ``'inferno'`` color map.
+
+    >>> import pyvista as pv
+    >>> lut = pv.LookupTable('inferno', n_values=32)
+    >>> lut  # doctest:+SKIP
+    LookupTable (0x7ff3c053f3a0)
+      Table Range:                (0.0, 1.0)
+      N Values:                   32
+      Above Range Color:          None
+      Below Range Color:          None
+      NAN Color:                  Color(name='maroon', hex='#800000ff')
+      Log Scale:                  False
+      Color Map:                  "inferno"
+    >>> lut.plot()
+
+    """
+
+    _nan_color_set = False
+    _cmap = None
+    _values_manual = False
+
+    def __init__(self, cmap=None, n_values=256, flip=False):
```

Comment:
Can we support all properties as kwargs in the `__init__`? I'd expect to do something like the following in a single line:

```py
lut = pv.LookupTable(cmap='bwr', table_range=(2,20))
```